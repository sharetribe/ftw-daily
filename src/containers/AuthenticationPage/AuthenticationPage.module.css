@import '../../styles/customMediaQueries.css';
@import '../../styles/propertySets.css';

.layoutWrapperMain {
  min-height: calc(100vh - var(--topbarHeight));

  @media (--viewportMedium) {
    min-height: calc(100vh - var(--topbarHeightDesktop));
  }
}

.root {
  @apply --marketplaceModalRootStyles;

  @media (--viewportMedium) {
    @apply --backgroundImage;
  }
}

.content {
  @apply --marketplaceModalBaseStyles;
  padding-top: 29px;
}

/* ================ Tabs & Form ================ */

.tabs {
  @media (--viewportMedium) {
    margin-top: 6px;
  }
}

.tab {
  @apply --marketplaceModalTitleStyles;
  margin-top: 0;
  margin-bottom: 0;

  @media (--viewportMedium) {
    margin-top: 0;
  }
}

.form {
  margin-top: 24px;

  @media (--viewportMedium) {
    margin-top: 30px;
    padding-top: 2px;
  }
}

.loginForm {
  margin-top: 24px;

  /* We don't want the form to take the whole space so that on mobile view
  the social login buttons will be after the sign up button
  and not in the bottom of the page.
  We also need to add flex-basis value so that Safari will show the
  whole form correctly.  */
  flex-grow: 0;
  flex-basis: 260px;

  @media (--viewportMedium) {
    flex-basis: 330px;
    margin-top: 30px;
    padding-top: 2px;
  }
}

.signupForm {
  margin-top: 24px;

  /* We don't want the form to take the whole space so that on mobile view
  the social login buttons will be after the sign up button
  and not in the bottom of the page.
  We also need to add flex-basis value so that Safari will show the
  whole form correctly.  */
  flex-grow: 0;
  flex-basis: 340px;

  @media (--viewportMedium) {
    flex-basis: 410px;
    margin-top: 30px;
    padding-top: 2px;
  }
}

.bottomWrapper {
  @apply --marketplaceModalBottomWrapper;

  /* Align the helper links to the left since there isn't a Primary Button anymore */
  text-align: left;
}

/* Terms of Service modal*/

.termsWrapper {
  width: 100%;
  padding-top: 40px;

  @media (--viewportMedium) {
    width: 604px;
    padding-top: 11px;
  }
}

.termsHeading {
  @apply --marketplaceH1FontStyles;
  margin: 0 0 19px 0;

  @media (--viewportMedium) {
    margin: 0 0 19px 0;
  }
}

/* ================ Hide Top bar in screens smaller than 768px  ================ */

.hideOnMobile {
  display: none;

  @media (--viewportMedium) {
    display: block;
  }
}

/* ================ Close icon ================ */

.verifyClose {
  @apply --marketplaceModalCloseStyles;

  /* This is not a real modal, z-index should not be modal's z-index */
  z-index: 1;
}

.closeText {
  @apply --marketplaceModalCloseText;
}

.closeIcon {
  @apply --marketplaceModalCloseIcon;
}

/* ================ Typography ================ */

/* Title of the modal */
.modalTitle {
  @apply --marketplaceModalTitleStyles;
}

/* Paragraph for the Modal */
.modalMessage {
  @apply --marketplaceModalParagraphStyles;
}

/* Make the email pop */
.email {
  font-weight: var(--fontWeightHighlightEmail);
}

/* Helper links */
.modalHelperLink {
  @apply --marketplaceLinkStyles;
  @apply --marketplaceModalHelperLink;
}

/* Helper texts for the links, not the actual links */
.modalHelperText {
  @apply --marketplaceModalHelperText;
}

.modalIcon {
  @apply --marketplaceModalIconStyles;

  /* Align the icon and rest of the content in the modal. Sign & Login doesn't use an icon, maybe we should add one? */
  margin-top: 47px;

  @media (--viewportMedium) {
    margin-top: 0;
  }
}

/* ================ Errors ================ */

.error {
  @apply --marketplaceModalErrorStyles;
}

/* ================ Social logins & SSO ================ */

.signupWithIdpTitle {
  @apply --marketplaceModalTitleStyles;
  margin-top: 0;
  margin-bottom: 0;
  padding-top: 16px;
  color: var(--matterColorDark);

  @media (--viewportMedium) {
    margin-top: 6px;
  }
}

.confirmInfoText {
  @apply --marketplaceH4FontStyles;
}

.buttonIcon {
  position: absolute;
  left: 0;
  margin-left: 20px;
}

.socialButtonWrapper {
  margin-bottom: 6px;
  @media (--viewportMedium) {
    margin-top: 8px;
  }
}

.socialButtonsOr {
  width: 100%;
  height: 32px;
  margin: 28px 0 20px 0;
  text-align: center;
  position: relative;
  background-color: var(--matterColorLight);

  &:after {
    content: '';
    width: 100%;
    border-bottom: solid 1px #d2d2d2;
    position: absolute;
    left: 0;
    top: 50%;
    z-index: 1;
  }
}

@media (--viewportMedium) {
  .socialButtonsOr {
    height: 34px;
    margin: 15px 0;
  }
}

.socialButtonsOrText {
  background-color: var(--matterColorLight);
  width: auto;
  display: inline-block;
  z-index: 3;
  padding: 0 20px 0 20px;
  position: relative;
  margin: 0;
}
