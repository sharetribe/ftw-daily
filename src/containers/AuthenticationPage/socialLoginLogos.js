import React from 'react';

// This is Facebook's logo, you are not allowed to change its color
export const FacebookLogo = (
  <svg width="20" height="20" xmlns="http://www.w3.org/2000/svg">
    <path
      d="M7.89.214C4.055 1.047 1.005 4.13.205 7.947c-.734 3.45.533 7.283 3.166 9.6.967.85 3.2 2.033 4.15 2.183l.617.1v-6.883H5.806v-3h2.283l.083-1.633c.134-2.417.717-3.534 2.3-4.25.617-.284 1.034-.35 2.3-.334.85.017 1.617.084 1.7.134.1.05.167.7.167 1.433v1.317h-.983c-1.484 0-1.75.283-1.817 1.983l-.067 1.35h1.45c1.284 0 1.434.033 1.35.283-.05.167-.133.667-.2 1.134-.216 1.55-.25 1.583-1.483 1.583h-1.083V19.914l.866-.234c1.684-.433 2.984-1.216 4.4-2.633 2.067-2.067 2.9-4.1 2.9-7.017 0-3.166-1.2-5.75-3.616-7.766C14.106.38 10.772-.42 7.889.214z"
      fill="#1877F2"
      fillRule="nonzero"
    />
  </svg>
);

// This is Google's logo, you are not allowed to change its color
export const GoogleLogo = (
  <svg width="20" height="20" xmlns="http://www.w3.org/2000/svg">
    <g fill="none" fillRule="evenodd">
      <path
        d="M19.6 10.227c0-.709-.064-1.39-.182-2.045H10v3.868h5.382a4.6 4.6 0 01-1.996 3.018v2.51h3.232c1.891-1.742 2.982-4.305 2.982-7.35z"
        fill="#4285F4"
      />
      <path
        d="M10 20c2.7 0 4.964-.895 6.618-2.423l-3.232-2.509c-.895.6-2.04.955-3.386.955-2.605 0-4.81-1.76-5.595-4.123H1.064v2.59A9.996 9.996 0 0010 20z"
        fill="#34A853"
      />
      <path
        d="M4.405 11.9c-.2-.6-.314-1.24-.314-1.9 0-.66.114-1.3.314-1.9V5.51H1.064A9.996 9.996 0 000 10c0 1.614.386 3.14 1.064 4.49l3.34-2.59z"
        fill="#FBBC05"
      />
      <path
        d="M10 3.977c1.468 0 2.786.505 3.823 1.496l2.868-2.868C14.959.99 12.695 0 10 0 6.09 0 2.71 2.24 1.064 5.51l3.34 2.59C5.192 5.736 7.396 3.977 10 3.977z"
        fill="#EA4335"
      />
      <path d="M0 0h20v20H0z" />
    </g>
  </svg>
);
