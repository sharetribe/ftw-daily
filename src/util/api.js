// These helpers are calling FTW's own server-side routes
// so, they are not directly calling Marketplace API or Integration API.
// You can find these api endpoints from 'server/api/...' directory

import { types as sdkTypes, transit } from './sdkLoader';
import config from '../config';
import Decimal from 'decimal.js';

export const apiBaseUrl = () => {
  const port = process.env.REACT_APP_DEV_API_SERVER_PORT;
  const useDevApiServer = process.env.NODE_ENV === 'development' && !!port;

  // In development, the dev API server is running in a different port
  if (useDevApiServer) {
    return `http://localhost:${port}`;
  }

  // Otherwise, use the same domain and port as the frontend
  return `${window.location.origin}`;
};

// Application type handlers for JS SDK.
//
// NOTE: keep in sync with `typeHandlers` in `server/api-util/sdk.js`
export const typeHandlers = [
  // Use Decimal type instead of SDK's BigDecimal.
  {
    type: sdkTypes.BigDecimal,
    customType: Decimal,
    writer: v => new sdkTypes.BigDecimal(v.toString()),
    reader: v => new Decimal(v.value),
  },
];

const serialize = data => {
  return transit.write(data, { typeHandlers, verbose: config.sdk.transitVerbose });
};

const deserialize = str => {
  return transit.read(str, { typeHandlers });
};

const post = (path, body) => {
  const url = `${apiBaseUrl()}${path}`;
  const options = {
    method: 'POST',
    credentials: 'include',
    headers: {
      'Content-Type': 'application/transit+json',
    },
    body: serialize(body),
  };
  return window.fetch(url, options).then(res => {
    const contentTypeHeader = res.headers.get('Content-Type');
    const contentType = contentTypeHeader ? contentTypeHeader.split(';')[0] : null;

    if (res.status >= 400) {
      return res.json().then(data => {
        let e = new Error();
        e = Object.assign(e, data);

        throw e;
      });
    }
    if (contentType === 'application/transit+json') {
      return res.text().then(deserialize);
    } else if (contentType === 'application/json') {
      return res.json();
    }
    return res.text();
  });
};

// Fetch transaction line items from the local API endpoint.
//
// See `server/api/transaction-line-items.js` to see what data should
// be sent in the body.
export const transactionLineItems = body => {
  return post('/api/transaction-line-items', body);
};

// Initiate a privileged transaction.
//
// With privileged transitions, the transactions need to be created
// from the backend. This endpoint enables sending the booking data to
// the local backend, and passing that to the Marketplace API.
//
// See `server/api/initiate-privileged.js` to see what data should be
// sent in the body.
export const initiatePrivileged = body => {
  return post('/api/initiate-privileged', body);
};

// Transition a transaction with a privileged transition.
//
// This is similar to the `initiatePrivileged` above. It will use the
// backend for the transition. The backend endpoint will add the
// payment line items to the transition params.
//
// See `server/api/transition-privileged.js` to see what data should
// be sent in the body.
export const transitionPrivileged = body => {
  return post('/api/transition-privileged', body);
};

// Create user with identity provider (e.g. Facebook or Google)
//
// If loginWithIdp api call fails and user can't authenticate to Flex with idp
// we will show option to create a new user with idp.
// For that user needs to confirm data fetched from the idp.
// After the confirmation, this endpoint is called to create a new user with confirmed data.
//
// See `server/api/auth/createUserWithIdp.js` to see what data should
// be sent in the body.
export const createUserWithIdp = body => {
  return post('/api/auth/create-user-with-idp', body);
};
