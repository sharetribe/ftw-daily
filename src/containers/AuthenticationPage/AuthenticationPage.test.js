import React from 'react';
import { renderShallow } from '../../util/test-helpers';
import { fakeIntl } from '../../util/test-data';
import { AuthenticationPageComponent } from './AuthenticationPage';

const noop = () => null;

describe('AuthenticationPageComponent', () => {
  // We need to overwrite social login client ids before running the test
  // to make sure it's same in local environment and in CI
  beforeEach(() => {
    process.env = Object.assign(process.env, { REACT_APP_FACEBOOK_APP_ID: '' });
    process.env = Object.assign(process.env, { REACT_APP_GOOGLE_CLIENT_ID: '' });
  });

  it('matches snapshot', () => {
    const props = {
      history: { push: noop },
      location: { state: { from: '/protected' } },
      tab: 'login',
      isAuthenticated: false,
      authInProgress: false,
      scrollingDisabled: false,
      currentUserHasListings: false,
      onLogout: noop,
      onManageDisableScrolling: noop,
      submitLogin: noop,
      submitSignup: noop,
      intl: fakeIntl,
      sendVerificationEmailInProgress: false,
      onResendVerificationEmail: noop,
    };
    const tree = renderShallow(<AuthenticationPageComponent {...props} />);
    expect(tree).toMatchSnapshot();
  });
});

describe('AuthenticationPageComponent with Facebook login', () => {
  // We need to overwrite social login client ids before running the test
  // to make sure it's same in local environment and in CI
  beforeEach(() => {
    process.env = Object.assign(process.env, { REACT_APP_FACEBOOK_APP_ID: 'test' });
    process.env = Object.assign(process.env, { REACT_APP_GOOGLE_CLIENT_ID: '' });
  });

  it('matches snapshot', () => {
    const props = {
      history: { push: noop },
      location: { state: { from: '/protected' } },
      tab: 'login',
      isAuthenticated: false,
      authInProgress: false,
      scrollingDisabled: false,
      currentUserHasListings: false,
      onLogout: noop,
      onManageDisableScrolling: noop,
      submitLogin: noop,
      submitSignup: noop,
      intl: fakeIntl,
      sendVerificationEmailInProgress: false,
      onResendVerificationEmail: noop,
    };
    const tree = renderShallow(<AuthenticationPageComponent {...props} />);
    expect(tree).toMatchSnapshot();
  });
});

describe('AuthenticationPageComponent with Google login', () => {
  // We need to overwrite social login client ids before running the test
  // to make sure it's same in local environment and in CI
  beforeEach(() => {
    process.env = Object.assign(process.env, { REACT_APP_FACEBOOK_APP_ID: '' });
    process.env = Object.assign(process.env, { REACT_APP_GOOGLE_CLIENT_ID: 'test' });
  });

  it('matches snapshot', () => {
    const props = {
      history: { push: noop },
      location: { state: { from: '/protected' } },
      tab: 'login',
      isAuthenticated: false,
      authInProgress: false,
      scrollingDisabled: false,
      currentUserHasListings: false,
      onLogout: noop,
      onManageDisableScrolling: noop,
      submitLogin: noop,
      submitSignup: noop,
      intl: fakeIntl,
      sendVerificationEmailInProgress: false,
      onResendVerificationEmail: noop,
    };
    const tree = renderShallow(<AuthenticationPageComponent {...props} />);
    expect(tree).toMatchSnapshot();
  });
});

describe('AuthenticationPageComponent with Google and Facebook login', () => {
  // We need to overwrite social login client ids before running the test
  // to make sure it's same in local environment and in CI
  beforeEach(() => {
    process.env = Object.assign(process.env, { REACT_APP_FACEBOOK_APP_ID: 'test' });
    process.env = Object.assign(process.env, { REACT_APP_GOOGLE_CLIENT_ID: 'test' });
  });

  it('matches snapshot', () => {
    const props = {
      history: { push: noop },
      location: { state: { from: '/protected' } },
      tab: 'login',
      isAuthenticated: false,
      authInProgress: false,
      scrollingDisabled: false,
      currentUserHasListings: false,
      onLogout: noop,
      onManageDisableScrolling: noop,
      submitLogin: noop,
      submitSignup: noop,
      intl: fakeIntl,
      sendVerificationEmailInProgress: false,
      onResendVerificationEmail: noop,
    };
    const tree = renderShallow(<AuthenticationPageComponent {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
