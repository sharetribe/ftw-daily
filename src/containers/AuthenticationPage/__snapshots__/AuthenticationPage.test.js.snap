// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthenticationPageComponent matches snapshot 1`] = `
<Page
  schema={
    Object {
      "@context": "http://schema.org",
      "@type": "WebPage",
      "name": "AuthenticationPage.schemaTitleLogin",
    }
  }
  scrollingDisabled={false}
  title="AuthenticationPage.schemaTitleLogin"
>
  <LayoutSingleColumn
    className={null}
    rootClassName={null}
  >
    <LayoutWrapperTopbar
      className={null}
      rootClassName={null}
    >
      <withRouter(Connect(TopbarContainerComponent))
        className=""
      />
    </LayoutWrapperTopbar>
    <LayoutWrapperMain
      className="layoutWrapperMain"
      rootClassName={null}
    >
      <div
        className="root"
      >
        <div
          className="content"
        >
          <LinkTabNavHorizontal
            className="tabs"
            rootClassName={null}
            skin="light"
            tabClassName={null}
            tabRootClassName={null}
            tabs={
              Array [
                Object {
                  "linkProps": Object {
                    "name": "SignupPage",
                    "to": Object {
                      "state": Object {
                        "from": "/protected",
                      },
                    },
                  },
                  "selected": false,
                  "text": <h1
                    className="tab"
                  >
                    <Memo(MemoizedFormattedMessage)
                      id="AuthenticationPage.signupLinkText"
                    />
                  </h1>,
                },
                Object {
                  "linkProps": Object {
                    "name": "LoginPage",
                    "to": Object {
                      "state": Object {
                        "from": "/protected",
                      },
                    },
                  },
                  "selected": true,
                  "text": <h1
                    className="tab"
                  >
                    <Memo(MemoizedFormattedMessage)
                      id="AuthenticationPage.loginLinkText"
                    />
                  </h1>,
                },
              ]
            }
          />
          <LoginForm
            className="loginForm"
            inProgress={false}
            onSubmit={[Function]}
          />
        </div>
      </div>
      <injectIntl(ModalComponent)
        id="AuthenticationPage.tos"
        isOpen={false}
        onClose={[Function]}
        onManageDisableScrolling={[Function]}
        usePortal={true}
      >
        <div
          className="termsWrapper"
        >
          <h2
            className="termsHeading"
          >
            <MemoizedFormattedMessage
              id="AuthenticationPage.termsHeading"
            />
          </h2>
          <TermsOfService
            className={null}
            rootClassName={null}
          />
        </div>
      </injectIntl(ModalComponent)>
    </LayoutWrapperMain>
    <LayoutWrapperFooter
      className={null}
      rootClassName={null}
    >
      <injectIntl(Footer) />
    </LayoutWrapperFooter>
  </LayoutSingleColumn>
</Page>
`;

exports[`AuthenticationPageComponent with Facebook login matches snapshot 1`] = `
<Page
  schema={
    Object {
      "@context": "http://schema.org",
      "@type": "WebPage",
      "name": "AuthenticationPage.schemaTitleLogin",
    }
  }
  scrollingDisabled={false}
  title="AuthenticationPage.schemaTitleLogin"
>
  <LayoutSingleColumn
    className={null}
    rootClassName={null}
  >
    <LayoutWrapperTopbar
      className={null}
      rootClassName={null}
    >
      <withRouter(Connect(TopbarContainerComponent))
        className=""
      />
    </LayoutWrapperTopbar>
    <LayoutWrapperMain
      className="layoutWrapperMain"
      rootClassName={null}
    >
      <div
        className="root"
      >
        <div
          className="content"
        >
          <LinkTabNavHorizontal
            className="tabs"
            rootClassName={null}
            skin="light"
            tabClassName={null}
            tabRootClassName={null}
            tabs={
              Array [
                Object {
                  "linkProps": Object {
                    "name": "SignupPage",
                    "to": Object {
                      "state": Object {
                        "from": "/protected",
                      },
                    },
                  },
                  "selected": false,
                  "text": <h1
                    className="tab"
                  >
                    <Memo(MemoizedFormattedMessage)
                      id="AuthenticationPage.signupLinkText"
                    />
                  </h1>,
                },
                Object {
                  "linkProps": Object {
                    "name": "LoginPage",
                    "to": Object {
                      "state": Object {
                        "from": "/protected",
                      },
                    },
                  },
                  "selected": true,
                  "text": <h1
                    className="tab"
                  >
                    <Memo(MemoizedFormattedMessage)
                      id="AuthenticationPage.loginLinkText"
                    />
                  </h1>,
                },
              ]
            }
          />
          <LoginForm
            className="loginForm"
            inProgress={false}
            onSubmit={[Function]}
          />
          <div
            className="idpButtons"
          >
            <div
              className="socialButtonsOr"
            >
              <span
                className="socialButtonsOrText"
              >
                <MemoizedFormattedMessage
                  id="AuthenticationPage.or"
                />
              </span>
            </div>
            <div
              className="socialButtonWrapper"
            >
              <SocialLoginButton
                onClick={[Function]}
              >
                <span
                  className="buttonIcon"
                >
                  <svg
                    height="20"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M7.89.214C4.055 1.047 1.005 4.13.205 7.947c-.734 3.45.533 7.283 3.166 9.6.967.85 3.2 2.033 4.15 2.183l.617.1v-6.883H5.806v-3h2.283l.083-1.633c.134-2.417.717-3.534 2.3-4.25.617-.284 1.034-.35 2.3-.334.85.017 1.617.084 1.7.134.1.05.167.7.167 1.433v1.317h-.983c-1.484 0-1.75.283-1.817 1.983l-.067 1.35h1.45c1.284 0 1.434.033 1.35.283-.05.167-.133.667-.2 1.134-.216 1.55-.25 1.583-1.483 1.583h-1.083V19.914l.866-.234c1.684-.433 2.984-1.216 4.4-2.633 2.067-2.067 2.9-4.1 2.9-7.017 0-3.166-1.2-5.75-3.616-7.766C14.106.38 10.772-.42 7.889.214z"
                      fill="#1877F2"
                      fillRule="nonzero"
                    />
                  </svg>
                </span>
                <MemoizedFormattedMessage
                  id="AuthenticationPage.loginWithFacebook"
                />
              </SocialLoginButton>
            </div>
          </div>
        </div>
      </div>
      <injectIntl(ModalComponent)
        id="AuthenticationPage.tos"
        isOpen={false}
        onClose={[Function]}
        onManageDisableScrolling={[Function]}
        usePortal={true}
      >
        <div
          className="termsWrapper"
        >
          <h2
            className="termsHeading"
          >
            <MemoizedFormattedMessage
              id="AuthenticationPage.termsHeading"
            />
          </h2>
          <TermsOfService
            className={null}
            rootClassName={null}
          />
        </div>
      </injectIntl(ModalComponent)>
    </LayoutWrapperMain>
    <LayoutWrapperFooter
      className={null}
      rootClassName={null}
    >
      <injectIntl(Footer) />
    </LayoutWrapperFooter>
  </LayoutSingleColumn>
</Page>
`;

exports[`AuthenticationPageComponent with Google and Facebook login matches snapshot 1`] = `
<Page
  schema={
    Object {
      "@context": "http://schema.org",
      "@type": "WebPage",
      "name": "AuthenticationPage.schemaTitleLogin",
    }
  }
  scrollingDisabled={false}
  title="AuthenticationPage.schemaTitleLogin"
>
  <LayoutSingleColumn
    className={null}
    rootClassName={null}
  >
    <LayoutWrapperTopbar
      className={null}
      rootClassName={null}
    >
      <withRouter(Connect(TopbarContainerComponent))
        className=""
      />
    </LayoutWrapperTopbar>
    <LayoutWrapperMain
      className="layoutWrapperMain"
      rootClassName={null}
    >
      <div
        className="root"
      >
        <div
          className="content"
        >
          <LinkTabNavHorizontal
            className="tabs"
            rootClassName={null}
            skin="light"
            tabClassName={null}
            tabRootClassName={null}
            tabs={
              Array [
                Object {
                  "linkProps": Object {
                    "name": "SignupPage",
                    "to": Object {
                      "state": Object {
                        "from": "/protected",
                      },
                    },
                  },
                  "selected": false,
                  "text": <h1
                    className="tab"
                  >
                    <Memo(MemoizedFormattedMessage)
                      id="AuthenticationPage.signupLinkText"
                    />
                  </h1>,
                },
                Object {
                  "linkProps": Object {
                    "name": "LoginPage",
                    "to": Object {
                      "state": Object {
                        "from": "/protected",
                      },
                    },
                  },
                  "selected": true,
                  "text": <h1
                    className="tab"
                  >
                    <Memo(MemoizedFormattedMessage)
                      id="AuthenticationPage.loginLinkText"
                    />
                  </h1>,
                },
              ]
            }
          />
          <LoginForm
            className="loginForm"
            inProgress={false}
            onSubmit={[Function]}
          />
          <div
            className="idpButtons"
          >
            <div
              className="socialButtonsOr"
            >
              <span
                className="socialButtonsOrText"
              >
                <MemoizedFormattedMessage
                  id="AuthenticationPage.or"
                />
              </span>
            </div>
            <div
              className="socialButtonWrapper"
            >
              <SocialLoginButton
                onClick={[Function]}
              >
                <span
                  className="buttonIcon"
                >
                  <svg
                    height="20"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M7.89.214C4.055 1.047 1.005 4.13.205 7.947c-.734 3.45.533 7.283 3.166 9.6.967.85 3.2 2.033 4.15 2.183l.617.1v-6.883H5.806v-3h2.283l.083-1.633c.134-2.417.717-3.534 2.3-4.25.617-.284 1.034-.35 2.3-.334.85.017 1.617.084 1.7.134.1.05.167.7.167 1.433v1.317h-.983c-1.484 0-1.75.283-1.817 1.983l-.067 1.35h1.45c1.284 0 1.434.033 1.35.283-.05.167-.133.667-.2 1.134-.216 1.55-.25 1.583-1.483 1.583h-1.083V19.914l.866-.234c1.684-.433 2.984-1.216 4.4-2.633 2.067-2.067 2.9-4.1 2.9-7.017 0-3.166-1.2-5.75-3.616-7.766C14.106.38 10.772-.42 7.889.214z"
                      fill="#1877F2"
                      fillRule="nonzero"
                    />
                  </svg>
                </span>
                <MemoizedFormattedMessage
                  id="AuthenticationPage.loginWithFacebook"
                />
              </SocialLoginButton>
            </div>
            <div
              className="socialButtonWrapper"
            >
              <SocialLoginButton
                onClick={[Function]}
              >
                <span
                  className="buttonIcon"
                >
                  <svg
                    height="20"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      fill="none"
                      fillRule="evenodd"
                    >
                      <path
                        d="M19.6 10.227c0-.709-.064-1.39-.182-2.045H10v3.868h5.382a4.6 4.6 0 01-1.996 3.018v2.51h3.232c1.891-1.742 2.982-4.305 2.982-7.35z"
                        fill="#4285F4"
                      />
                      <path
                        d="M10 20c2.7 0 4.964-.895 6.618-2.423l-3.232-2.509c-.895.6-2.04.955-3.386.955-2.605 0-4.81-1.76-5.595-4.123H1.064v2.59A9.996 9.996 0 0010 20z"
                        fill="#34A853"
                      />
                      <path
                        d="M4.405 11.9c-.2-.6-.314-1.24-.314-1.9 0-.66.114-1.3.314-1.9V5.51H1.064A9.996 9.996 0 000 10c0 1.614.386 3.14 1.064 4.49l3.34-2.59z"
                        fill="#FBBC05"
                      />
                      <path
                        d="M10 3.977c1.468 0 2.786.505 3.823 1.496l2.868-2.868C14.959.99 12.695 0 10 0 6.09 0 2.71 2.24 1.064 5.51l3.34 2.59C5.192 5.736 7.396 3.977 10 3.977z"
                        fill="#EA4335"
                      />
                      <path
                        d="M0 0h20v20H0z"
                      />
                    </g>
                  </svg>
                </span>
                <MemoizedFormattedMessage
                  id="AuthenticationPage.loginWithGoogle"
                />
              </SocialLoginButton>
            </div>
          </div>
        </div>
      </div>
      <injectIntl(ModalComponent)
        id="AuthenticationPage.tos"
        isOpen={false}
        onClose={[Function]}
        onManageDisableScrolling={[Function]}
        usePortal={true}
      >
        <div
          className="termsWrapper"
        >
          <h2
            className="termsHeading"
          >
            <MemoizedFormattedMessage
              id="AuthenticationPage.termsHeading"
            />
          </h2>
          <TermsOfService
            className={null}
            rootClassName={null}
          />
        </div>
      </injectIntl(ModalComponent)>
    </LayoutWrapperMain>
    <LayoutWrapperFooter
      className={null}
      rootClassName={null}
    >
      <injectIntl(Footer) />
    </LayoutWrapperFooter>
  </LayoutSingleColumn>
</Page>
`;

exports[`AuthenticationPageComponent with Google login matches snapshot 1`] = `
<Page
  schema={
    Object {
      "@context": "http://schema.org",
      "@type": "WebPage",
      "name": "AuthenticationPage.schemaTitleLogin",
    }
  }
  scrollingDisabled={false}
  title="AuthenticationPage.schemaTitleLogin"
>
  <LayoutSingleColumn
    className={null}
    rootClassName={null}
  >
    <LayoutWrapperTopbar
      className={null}
      rootClassName={null}
    >
      <withRouter(Connect(TopbarContainerComponent))
        className=""
      />
    </LayoutWrapperTopbar>
    <LayoutWrapperMain
      className="layoutWrapperMain"
      rootClassName={null}
    >
      <div
        className="root"
      >
        <div
          className="content"
        >
          <LinkTabNavHorizontal
            className="tabs"
            rootClassName={null}
            skin="light"
            tabClassName={null}
            tabRootClassName={null}
            tabs={
              Array [
                Object {
                  "linkProps": Object {
                    "name": "SignupPage",
                    "to": Object {
                      "state": Object {
                        "from": "/protected",
                      },
                    },
                  },
                  "selected": false,
                  "text": <h1
                    className="tab"
                  >
                    <Memo(MemoizedFormattedMessage)
                      id="AuthenticationPage.signupLinkText"
                    />
                  </h1>,
                },
                Object {
                  "linkProps": Object {
                    "name": "LoginPage",
                    "to": Object {
                      "state": Object {
                        "from": "/protected",
                      },
                    },
                  },
                  "selected": true,
                  "text": <h1
                    className="tab"
                  >
                    <Memo(MemoizedFormattedMessage)
                      id="AuthenticationPage.loginLinkText"
                    />
                  </h1>,
                },
              ]
            }
          />
          <LoginForm
            className="loginForm"
            inProgress={false}
            onSubmit={[Function]}
          />
          <div
            className="idpButtons"
          >
            <div
              className="socialButtonsOr"
            >
              <span
                className="socialButtonsOrText"
              >
                <MemoizedFormattedMessage
                  id="AuthenticationPage.or"
                />
              </span>
            </div>
            <div
              className="socialButtonWrapper"
            >
              <SocialLoginButton
                onClick={[Function]}
              >
                <span
                  className="buttonIcon"
                >
                  <svg
                    height="20"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      fill="none"
                      fillRule="evenodd"
                    >
                      <path
                        d="M19.6 10.227c0-.709-.064-1.39-.182-2.045H10v3.868h5.382a4.6 4.6 0 01-1.996 3.018v2.51h3.232c1.891-1.742 2.982-4.305 2.982-7.35z"
                        fill="#4285F4"
                      />
                      <path
                        d="M10 20c2.7 0 4.964-.895 6.618-2.423l-3.232-2.509c-.895.6-2.04.955-3.386.955-2.605 0-4.81-1.76-5.595-4.123H1.064v2.59A9.996 9.996 0 0010 20z"
                        fill="#34A853"
                      />
                      <path
                        d="M4.405 11.9c-.2-.6-.314-1.24-.314-1.9 0-.66.114-1.3.314-1.9V5.51H1.064A9.996 9.996 0 000 10c0 1.614.386 3.14 1.064 4.49l3.34-2.59z"
                        fill="#FBBC05"
                      />
                      <path
                        d="M10 3.977c1.468 0 2.786.505 3.823 1.496l2.868-2.868C14.959.99 12.695 0 10 0 6.09 0 2.71 2.24 1.064 5.51l3.34 2.59C5.192 5.736 7.396 3.977 10 3.977z"
                        fill="#EA4335"
                      />
                      <path
                        d="M0 0h20v20H0z"
                      />
                    </g>
                  </svg>
                </span>
                <MemoizedFormattedMessage
                  id="AuthenticationPage.loginWithGoogle"
                />
              </SocialLoginButton>
            </div>
          </div>
        </div>
      </div>
      <injectIntl(ModalComponent)
        id="AuthenticationPage.tos"
        isOpen={false}
        onClose={[Function]}
        onManageDisableScrolling={[Function]}
        usePortal={true}
      >
        <div
          className="termsWrapper"
        >
          <h2
            className="termsHeading"
          >
            <MemoizedFormattedMessage
              id="AuthenticationPage.termsHeading"
            />
          </h2>
          <TermsOfService
            className={null}
            rootClassName={null}
          />
        </div>
      </injectIntl(ModalComponent)>
    </LayoutWrapperMain>
    <LayoutWrapperFooter
      className={null}
      rootClassName={null}
    >
      <injectIntl(Footer) />
    </LayoutWrapperFooter>
  </LayoutSingleColumn>
</Page>
`;
